### DISCUSSION TITLE
Composer.json requires php ^8.1 but Renovate runs the project with v7.4

### DISCUSSION THREAD

**Discussion by simPod** [renovatebot/renovate#14754] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate/renovate:32

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

14.8.2-ee

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Composer.json (and lockfile)

```json
{
    "require": {
        "php": "^8.1",
```

requires php 8.1+

but is run with 7.4.28

### Relevant debug logs

<details><summary>Logs</summary>

```
- symfony/process v6.0.5 requires php >=8.0.2 -> your php version (7.4.28) does not satisfy that requirement
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by simPod** [comment]
I checked source code 

https://github.com/renovatebot/renovate/blob/56c8616b9af4383b9b98a421a16bd185b6b10465/lib/modules/manager/composer/utils.ts#L75

and I guess it does not work for `^8.1`, since it extracts `^8.1` which resolves to following docker image tag `php:^8.1` which is invalid?

**Comment by viceice** [comment]
https://github.com/renovatebot/renovate/blob/56c8616b9af4383b9b98a421a16bd185b6b10465/lib/modules/manager/composer/artifacts.ts#L116-L119

We are using composer versioning to find a matching docker tag from https://hub.docker.com/r/renovate/php/tags

**Comment by rarkins** [comment]
Isn't this because the full image is being used, and for dynamic languages you need binarySource=install docker socket?

**Comment by simPod** [comment]
I've found some mention of it here https://github.com/renovatebot/renovate/issues/2355#issuecomment-1010015101

It says to use slim.

Let me set `binarySource: 'docker'`

**Comment by viceice** [comment, marked as answer]
Converted to discussion, as this is expected behavior with full image currently.

- ref renovatebot/docker-renovate-full#406

**Reply by simPod** [reply]
@viceice binarySource install still goes with php 7.4 🤔

**Reply by viceice** [reply]
I know, that's why the issue is still open. only a few tools are yet compatible

Here are the supported tools:
https://github.com/renovatebot/renovate/blob/394bb29f3f0f286e2249247f5568a8defd2ed679/lib/util/exec/buildpack.ts#L12-L54

**Reply by viceice** [reply]
So the options for you are
* use slim image with docker mode
* build / extend renovate docker image
